test/fail_compilation/test19961.d

```diff
@@ -0,0 +1,46 @@
+/*
+REQUIRED_ARGS:
+PERMUTE_ARGS:
+TEST_OUTPUT:
+---
+fail_compilation/test19961.d(20): Error: `immutable` delegate `test15306.main.__dgliteral1` cannot access mutable data `i`
+fail_compilation/test19961.d(28): Error: `const` delegate test15306.main.__dgliteral2 cannot access mutable data `i`
+fail_compilation/test19961.d(35): Error: `inout` delegate test15306.main.__dgliteral2 cannot access mutable data `i`
+---
+*/
+
+// https://issues.dlang.org/show_bug.cgi?id=15306
+// https://issues.dlang.org/show_bug.cgi?id=19961
+void main()
+{
+    // immutable cannot access mutable
+    int i = 42;
+    auto dg1 = delegate void() immutable
+    {
+        i++;
```

Comment:
This should be an error, `i` is immutable in this scope.

Reply:
It is, this is a fail_compilation test.